docs/design/nvidia-gpu-operator.md

```diff
@@ -0,0 +1,19 @@
+# Nvidia GPU Operator Integration
+
+## Background
+Nvidia have a standard instructions on how to integrate their GPUs and drivers into a Kubernetes Cluster with GPU-Operator. GPU scheduling is required many HPC applications, particularly multi-node multi-gpu deep learning training.
+
+## Motivation
+Exclusive GPU sharing currently doesn't take advantage of the work already done by Nvidia's GPU operator in finding GPUs and adding node annotations. Using these annotations foregoes the effort of running our own device plugins for device discovery. This also has the advantage of forcing users to only be exposed to the scheduled of GPUs in their container rather than asking them to use the environment variables that are annotated by previous implementations.
+
+## Goals
+ - Support for exclusive access scheduling with based on the standard annotations provided by Nvidia.
+ - Limited user effort other than following Nvidia instructions for creating gpu cluster and changing volcano-scheduler-configmap.
+ - Add this feature with minimised required changes to existing codebase.
+
+## Non-Goals
+ - Support for gpu-sharing and mixed node annotation support (nodes should have the same type of GPU on them)
+
```

Comment:
hm... what are we going to do for this proposal?

Reply:
Using requests should always gives exclusive access to that resource. Without MIG there's no real way to properly allocate fractions of a card exclusively. The current "gpu sharing" doesn't actually enforce gpu memory sharing limits, any user can just request 1MB but use the whole card, nor does this take into consideration compute util vs memory util.

Requesting a specific card on a mixed card node would be possible but would not integrate with gpu operator  resource request handling. Would have to go back to writing custom annotators and allocators.